packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -357,39 +359,48 @@ export class FileStorage implements DStore {
     return { notes: allNotes, errors };
   }
 
-  async _addBacklinks({
+  /** Adds backlinks mutating 'allNotes' argument in place. */
+  private _addBacklinks({
     notesWithLinks,
     allNotes,
   }: {
     notesWithLinks: NoteProps[];
     allNotes: NoteProps[];
-    allNotesCache: NotesCacheAll;
-  }) {
-    return _.map(notesWithLinks, async (noteFrom) => {
+  }): void {
+    const ctx = "_addBacklinks:ext";
+    const start = process.hrtime();
+
+    this._addBacklinksImpl(allNotes, notesWithLinks);
+
+    const duration = getDurationMilliseconds(start);
+    this.logger.info({ ctx, duration });
+  }
+
+  private _addBacklinksImpl(
```

Comment:
since these are both private, probably fine to just merge into a single `_addBacklinks()` method - but I'll leave it up to you.